include/artifact-uploader.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2017 Prism Skylabs
+ */
+#ifndef CONNECT_SDK_ARTIFACT_UPLOADER_H
+#define CONNECT_SDK_ARTIFACT_UPLOADER_H
+
+#include "domain-types.h"
+#include "boost/shared_ptr.hpp"
+#include "public-util.h"
+
+namespace prism
+{
+namespace connect
+{
+
+class Client;
+// Usage example
+// namespace prc = prism::connect;
+// void configCallback(prc::Client& client)
+// {
+//     client.setLogFlags(prc::Client::LOG_INPUT | prc::Client::LOG_INPUT_JSON);
+//     client.setConnectionTimeoutMs(5000);
+//     client.setLowSpeed(5);
+// }
+//
+// prc::ArtifactUploader::Configuration uploaderConfig("<apiRoot>", "<apiToken>",
+//      "camOne", 32);
+// prc::ArtifactUploader uploader;
+// prc::Status status = uploader.init(uploaderConfig, configCallback);
+// if (status.isError())
+//     abort();
+// uploader.uploadBackground(<timestamp>, PayloadAu::makeByMovingData(<data>, <mimeType>));
+
+//class PayloadAu;
+//typedef boost::shared_ptr<PayloadAu> PayloadAuPtr;
+
+// TODO: other options for class name: SmartPayload, PayloadV2, PayloadTwo
+// This class is carefully designed to take ownership and clean data
+// after use.
+// The moment, when data is copied or stolen is well-defined: in a call
+// to make* static method.
+// The moment, when owned data or file is deleted is uknknown: whenever
+// PayloadAu instance is destroyed.
+// Copying and assignment are explicitly prohibited to prevent unintended
+// losing of ownership or data copying.
+// This class is intentionally made distinct from Payload struct to keep
+// current interface using Payload unchanged.
+class PayloadAu : private boost::noncopyable
```

Comment:
Nope. And there is a comment explaining that. So proposals on naming are welcome.